### ISSUE TITLE
dd-wrt test build by Richard

### ISSUE THREAD

**Issue by jfrux** [jfrux/dd-wrt-bootstrap-theme#2]
Richard has a test build of dd-wrt that allows for including custom css / js files and we're going to see if we can get it implemented in that test build.

**Comment by jfrux** [comment]
I have to ask... are you using Grunt to build the final files first before using them? or... just using the application.css found in this repo?

**Comment by ddwrtkong** [comment]
Yes I also recreated the js and css files:

~/Temp/bootstrap> grunt
Running "clean:dist" (clean) task
Cleaning assets/css/application.css...OK
Cleaning assets/js/application.js...OK

Running "less:dist" (less) task
File assets/css/application.css created.

Running "uglify:dist" (uglify) task
File "assets/js/application.js" created.

Done, without errors.

**Comment by jfrux** [comment]
Hmmm!

Any way to expose the test device to the web so I can poke around or is that not possible?

**Comment by ddwrtkong** [comment]
Yes should be possible, let me setup port forwards, as this device is not directly attached to the internet.

**Comment by jfrux** [comment]
I'm thinking all we need to do is move the application.js file to the bottom just before the </body> and it should work.

**Comment by jfrux** [comment]
Here is what my `dotjs` plugin JS file looks like which helps with development quick changes of the files locally...

``` javascript
var $css = $("<link></link>").attr({
  rel:"stylesheet",
  type:"text/css",
  href:"http://ddwrt.dev/assets/css/application.css"
});

var $js = $("<script></script>").attr({
  type:"text/javascript",
  src:"http://ddwrt.dev/assets/js/application.js"
});

$("head").append($css);
$("body").append($js);
```

**Comment by ddwrtkong** [comment]
Cool it works. I added it real quick to the bottom of the management page, since body tag is included in each page.  I have to check how we can improve this. I don't want to insert the script tag to each page.

**Comment by jfrux** [comment]
I will make it work for the 

```
<head></head>
```

 for you okay?
One moment.
Thanks!

**Comment by ddwrtkong** [comment]
That would be awesome. As this requires only a few lines of code.

**Comment by jfrux** [comment]
Okay, checkout the latest code I pushed.

application.js is now wrapped in a `$(document).ready(function() {});` so it should work within the 
`<head></head>` tags

**Comment by ddwrtkong** [comment]
Perfect, works like a charm! So now I just have to complete the integration, normally we compress and include the js/css files into the www bin for faster loading. 
Then I can commit it to dd-wrt repo, this way people give their feedback once the next builds are out. 

How should we name the theme?

**Comment by jfrux** [comment]
I'm not certain what to name it...?
Maybe that's a subject for the forums?

I have another issue I'm going to add to this repo regarding how we can make it easy for people to get updates for the theme without having to do too much technical jargon. :)

Issue for updating #3

**Comment by jfrux** [comment]
Maybe name it Next Generation [alpha] or [beta] or something...

**Comment by ddwrtkong** [comment]
I think It should be possible to add a symlink. If a certain style is selected it will pull the css and js from lets say /jffs/theme...
As for the main integration I would just add and update it in the repo from time to time. 

Responsive might be a nice name for it too.

**Comment by antoniobenedetti** [comment]
No test build yet? :cry: 
I'm currently running SVN revision 24461 but i don't see it. Did you commit it to DD-WRT repo @ddwrtkong ?

**Comment by jfrux** [comment]
Unfortunately my time is currently taken at the moment and I have not been equipped to prioritize this sadly…
I’m excited for it too, I just wish I could break away long enough to tweak things a bit and get it cleaned up.

**Comment by ddwrtkong** [comment]
No I did not commit it yet, as it is broken, the theme works and looks fine, but once it is enabled you cannot apply/save settings and thus cannot change back to a different theme only through ssh. Thus I also deactivated it in my builds, although it is included.

**Comment by antoniobenedetti** [comment]
Can you finish it guys please? Or is the project abandoned? :/
@joshuairl @ddwrtkong

**Comment by xZero707** [comment]
Unfortunately, it seems like project is abandoned, as there are no recent updates.

**Comment by jfrux** [comment]
@antoniobenedetti @xZero707 I have yet to have extra time.  This was a pet project I was working on during my time using dd-wrt.

Sadly, what dd-wrt really needs is a complete overhaul of the UI written in a much stronger JS framework such as Ember.  It'll be a fight to maintain without a solid framework to build off of.

I'm definitely going to keep this in the back of my mind though.

**Comment by xZero707** [comment]
I would like to help, but I'm not so good with javascript, I'm back-end dev/sys admin so JS never got to my way.
This project is great, and DD-WRT definitely need complete rebuild of UI, the default one is ugly and non-responsive... BUT! If they don't want to update, why would you care and work on this? You shouldn't.
I tuned many routers, for many people, and I was always giving DD-WRT advantage, however now I considering OpenWRT, as they actually care about important things like GUI is.
